boringssl: recognise every CPU the SDK is built for (loongarch64, s390x, ppc,
hexagon, big-endian ARM/MIPS) instead of #error. None of them get assembly;
arm64ec, which also defines __x86_64__, is matched first as the AArch64 it is.

--- a/src/boringssl/src/include/openssl/target.h
+++ b/src/boringssl/src/include/openssl/target.h
@@ -22,22 +22,25 @@
 // <openssl/base.h> and <openssl/asm_base.h>. Prefer to include those headers
 // instead.
 
-#if defined(__x86_64) || defined(_M_AMD64) || defined(_M_X64)
+#if defined(__arm64ec__) || defined(_M_ARM64EC)
+#define OPENSSL_64_BIT
+#define OPENSSL_AARCH64
+#elif defined(__x86_64) || defined(_M_AMD64) || defined(_M_X64)
 #define OPENSSL_64_BIT
 #define OPENSSL_X86_64
 #elif defined(__x86) || defined(__i386) || defined(__i386__) || defined(_M_IX86)
 #define OPENSSL_32_BIT
 #define OPENSSL_X86
-#elif defined(__AARCH64EL__) || defined(_M_ARM64)
+#elif defined(__AARCH64EL__) || defined(__AARCH64EB__) || defined(_M_ARM64)
 #define OPENSSL_64_BIT
 #define OPENSSL_AARCH64
-#elif defined(__ARMEL__) || defined(_M_ARM)
+#elif defined(__ARMEL__) || defined(__ARMEB__) || defined(_M_ARM)
 #define OPENSSL_32_BIT
 #define OPENSSL_ARM
-#elif defined(__MIPSEL__) && !defined(__LP64__)
+#elif (defined(__MIPSEL__) || defined(__MIPSEB__)) && !defined(__LP64__)
 #define OPENSSL_32_BIT
 #define OPENSSL_MIPS
-#elif defined(__MIPSEL__) && defined(__LP64__)
+#elif (defined(__MIPSEL__) || defined(__MIPSEB__)) && defined(__LP64__)
 #define OPENSSL_64_BIT
 #define OPENSSL_MIPS64
 #elif defined(__riscv) && __SIZEOF_POINTER__ == 8
@@ -53,6 +56,10 @@
 #elif defined(__asmjs__)
 #define OPENSSL_32_BIT
 #elif defined(__myriad2__)
+#define OPENSSL_32_BIT
+#elif defined(__loongarch64) || defined(__s390x__) || defined(__powerpc64__)
+#define OPENSSL_64_BIT
+#elif defined(__powerpc__) || defined(__hexagon__)
 #define OPENSSL_32_BIT
 #else
 // The list above enumerates the platforms that BoringSSL supports. For these
